src/server/connection.rs

```diff
@@ -1841,6 +1865,15 @@ impl Connection {
                                 },
                             );
                         }
+                        if !tfa.hwid.is_empty() && Self::enable_trusted_devices() {
+                            Config::add_trusted_device(TrustedDevice {
+                                hwid: tfa.hwid,
+                                time: hbb_common::get_time(),
+                                id: self.lr.my_id.clone(),
+                                name: self.lr.my_name.clone(),
+                                platform: self.lr.my_platform.clone(),
+                            });
+                        }
```

Comment:
Make sure not added again and again.

Reply:
By removing the old one.
```
pub fn add_trusted_device(device: TrustedDevice) {
        let mut devices = Self::get_trusted_devices();
        devices.retain(|d| d.hwid != device.hwid);
        devices.push(device);
        Self::set_trusted_devices(devices);
    }
```

Reply:
If no change, why still call `set_trusted_devices`? wasting io.

Reply:
`time` may be updated. This can happens when `Enable trusted devices` is disabled, then being connected to,  so hwid will not be sent, then option enabled.Durin one device is trusted and option is enabled, will not go to this line.